highfidelity/hifi
# Case 22379: Run-time switch for deferred/forward rendering

https://highfidelity.manuscript.com/f/cases/22379/Run-time-switch-for-Deferred-Forward-rendering

Test plan:
- Rendering should look the same as master on Quest and Desktop.
- On Desktop (only), you can switch between Forward and Deferred rendering by running this in the console:
    - Forward: `Render.getConfig("RenderMainView.DeferredForwardSwitch").branch = 1`
    - Deferred: `Render.getConfig("RenderMainView.DeferredForwardSwitch").branch = 0`
- In Deferred mode, you'll see things like shadows, haze, and local lights.  In Forward mode, you won't see those things and might see other rendering differences.
- In deferred mode, run [this](https://gist.githubusercontent.com/SamGondelman/1b7418c416df22ddc84683b72e23a2fd/raw/fc5f092e8ca728e2c7a78c73c751d6f936c13b58/RenderLayer.js).  You'll see something like this:
![image](https://user-images.githubusercontent.com/7650116/57546139-f34b9a80-7310-11e9-8ada-79612564067b.png)
- Switch to forward mode and nothing (/very little) should change.
- Press "u" to make all of the entities (except the model, polyline, and material) slightly transparent.  They should still look the same in forward/deferred.
- Press "o" to cycle through the 3 render layers.  In the last one, the HUD layer, everything will appear slightly different (color grading and AA), but they should look the same between forward and deferred.
- Known issues (on master already):
    - the polyvox entity doesn't seem to render
    - when you toggle the transparency, the particle gets a speedSpread of 1?
- The blue and orange teleport parabolas should render in both deferred and forward modes.
